File: solutions/553/main.py
from typing import *

class Solution:
    def optimalDivision(self, nums: List[int]) -> str:
        result = ""
        result += str(nums[0])
        if len(nums) == 1: 
            return result 
        result += "/"
        if len(nums) == 2: 
            result += str(nums[1])
            return result
        result += "("
        for i in range(1, len(nums)): 
            result += str(nums[i])
            if i != len(nums)-1:
                result += "/"
            else: 
                result += ")"
        return result 

if __name__ == "__main__": 
    nums = [1000,100,10,2]
    s = Solution()
    assert s.optimalDivision(nums) == "1000/(100/10/2)"

File: solutions/553/main.rs
struct Solution {}

// ref https://leetcode-cn.com/problems/optimal-division/solution/rust-553-zui-you-chu-fa-by-fffzlfk-k7y4/
impl Solution {
    pub fn optimal_division(nums: Vec<i32>) -> String {
        match nums.len() {
            1 => nums[0].to_string(), 
            2 => format!("{}/{}", nums[0], nums[1]), 
            _ => format!("{}/({})",
                    nums[0], 
                    nums[1..]
                        .iter()
                        .map(|num| num.to_string())
                        .collect::<Vec<String>>()
                        .join("/")
                    ), 
        }
    }
}

fn main() {
    assert_eq!(Solution::optimal_division(vec![1000,100,10,2]), "1000/(100/10/2)"); 
}